lib/active_record_extended/query_methods/with_cte.rb

```diff
@@ -215,4 +235,11 @@ def add_materialized_modifier(expression, cte, name)
   end
 end
 
-ActiveRecord::Relation.prepend(ActiveRecordExtended::QueryMethods::WithCTE)
+# Only prepend the WithCTE module if CTE functionality is not disabled for Rails 7.2+
+unless ActiveRecordExtended::CteDeprecationConfiguration.should_disable_cte?
+  ActiveRecord::Relation.prepend(ActiveRecordExtended::QueryMethods::WithCTE)
+end
+# Only prepend the WithCTE module if CTE functionality is not disabled for Rails 7.2+
+unless ActiveRecordExtended::CteDeprecationConfiguration.should_disable_cte?
+  ActiveRecord::Relation.prepend(ActiveRecordExtended::QueryMethods::WithCTE)
+end
```

Comment:
This will not work. You are going to run into the simple fact that this file is executed the moment the gem is loaded, giving no time to evaluate any modifications to the configuration.